python_files/Rules.py

```
@@ -0,0 +1,102 @@
+
+    def rule5_large_transaction(self, threshold=100000):
+        return self.df[self.df['Amount'] > threshold]['Transaction_Id'].tolist()
+
+    def rule6_high_frequency_same_source(self, max_txn=5):
+        flagged = []
+        for _, group in self.df.groupby('IMEI'):
+            if len(group) > max_txn:
+                flagged.extend(group['Transaction_Id'].tolist())
+        return flagged
+
+    def rule7_unusual_dest_accounts(self, unusual=['XYZ123', 'PQR999']):
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)